pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -128,23 +136,27 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 			}
 			replicas[i] = newStatefulSetPod(set, i)
 		}
-		// If we find a Pod that has not been created we create the Pod immediately and return
+		// If we find a Pod that has not been created we create the Pod
 		if !isCreated(replicas[i]) {
-			return ssc.podControl.CreateStatefulPod(set, replicas[i])
+			if err := ssc.podControl.CreateStatefulPod(set, replicas[i]); err != nil {
+				return err
+			}
+			// if the set does not allow bursting, return immediately
+			if monotonic {
+				return nil
+			}
```

Comment:
I mean
```go
if !isCreated(replicas[i]) {
   if err := ssc.podControl.CreateStatefulPod(set, replicas[i]); err != nil {
		return err
	}
   if monotonic {
		return nil
	}
   continue
}
```
since the replica you just created is not going to fall under any of the following cases.